lib/noid/minter.rb

```diff
@@ -1,19 +1,24 @@
 module Noid
   class Minter
-    attr_reader :seed, :seq, :template
+    attr_reader :template, :seq
     attr_writer :counters
 
     def initialize(options = {})
-      if options[:state]
-        # Only set the sequence ivar if this is a stateful minter
-        @seq = options[:seq]
-      else
-        @max_counters = options[:max_counters]
-        @after_mint = options[:after_mint]
-      end
-      @counters = options[:counters]
       @template = Template.new(options[:template])
-      seed(options[:seed], options[:seq])
+
+      @counters = options[:counters]
+      @max_counters = options[:max_counters]
+
+      # callback when an identifier is minted
+      @after_mint = options[:after_mint]
+
+      # used for random minters
+      @rand = options[:rand] if options[:rand].is_a? Random
```

Comment:
These three lines hurt me a bit. This unpacks the path through the lines more clearly to me:

```ruby
@rand = case options[:rand]
        when Random
          options[:rand]
        when String
          Marshal.load(options[:rand])
        else
          Random.new(options[:seed] || Random.new_seed)
        end
end
```